modules/programs/borgmatic.nix

```diff
@@ -168,7 +170,7 @@ let
       } // config.retention.extraConfig;
       consistency = removeNullValues { checks = config.consistency.checks; }
         // config.consistency.extraConfig;
-    };
+    } // config.extraConfig);
```

Comment:
It may be a bit unfortunate that this doesn't do a recursive update so `consistency.extraConfig = …` would give a different result compared to `extraConfig.consistency = …`.

Reply:
I think it is a very good point. Would you mind giving me a clue on how to implement that please?

Reply:
after discussing on Matrix, we decided to:

1. get rid of all the `extraConfig` options and only keep the top-level one
2. use `mkRenamedOptionModule` to warn the user about the deprecated API

Unfortunately, I've been unable to make it work. I pushed a WIP commit to show my latest try. Can someone please help?

Reply:
The latest issue I have is

```
error: You're trying to declare a value of type `list'
       rather than an attribute-set for the option
       `programs.borgmatic.backups.<name>.imports'!

       This usually happens if `programs.borgmatic.backups.<name>.imports' has option
       definitions inside that are not matched. Please check how to properly define
       this option by e.g. referring to `man 5 configuration.nix'!
(use '--show-trace' to show detailed location information)
```

Reply:
The immediate problem is that you're defining `imports` inside `options`, but it should be at the same level as `config` and `options`.

The real problem is that these `mkRemovedOptionModule`s will not work because of https://github.com/NixOS/nixpkgs/issues/96006. The most promising attempt to fix that is currently https://github.com/NixOS/nixpkgs/pull/207187.

Reply:
If I understand what @ncfavier just said, it seems impossible to deprecate the non-top-level `extraConfig` options for now. Having both a top-level `extraConfig` option and having non-top-level ones is messy. 

The only solution I see is to have no top-level `extraConfig` option. As a result, I suggest we introduce a top-level option for each of borgmatic's setting sections and add one `extraConfig` for each. If I'm not mistaken after reading the [example configuration file](https://torsion.org/borgmatic/docs/reference/config.yaml), this just means adding `output` and `hooks` top-level options.

What do you think @ncfavier and @rycee?

Reply:
I implemented this approach in https://github.com/nix-community/home-manager/pull/4108.